module/VuFind/src/VuFind/RecordDriver/AbstractBase.php

```diff
@@ -252,8 +252,10 @@ public function getRatingData(?int $userId = null)
         // Cache data since comments list may ask for same information repeatedly:
         $cacheKey = $userId ?? '-';
         if (!isset($this->ratingCache[$cacheKey])) {
-            $table = $this->getDbTable('Ratings');
-            $this->ratingCache[$cacheKey] = $table->getForResource(
+            $ratingsService = $this->getDbService(
```

Comment:
I would definitely like to decouple ratings logic from record drivers, similar to what I'd like to do for comments... but one step at a time! :-)